File: apps/mobile/ios/kini/KiniNativeSelectionRowManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(KiniNativeSelectionRowManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(disabled, BOOL)
RCT_EXPORT_VIEW_PROPERTY(onPress, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(selected, BOOL)
RCT_EXPORT_VIEW_PROPERTY(title, NSString)

@end

File: apps/mobile/ios/kini/KiniNativeSelectionRowManager.swift
import React
import UIKit

@objc(KiniNativeSelectionRowManager)
final class KiniNativeSelectionRowManager: RCTViewManager {
  override func view() -> UIView! {
    KiniNativeSelectionRowView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    true
  }
}
